tests/test.rs

```diff
@@ -58,6 +58,36 @@ fn version() {
         .is_match(version));
 }
 
+#[test]
+fn test_compressing_and_decompressing_small_input() {
+    // Input to be compressed and decompressed
+    let input: &[u8] = &[1, 2, 3];
+
+    let compressed = {
+        let mut output = vec![];
+        io::copy(
+            &mut &*input,
+            &mut write::Bz3Encoder::new(&mut output, 100 * KB).unwrap(),
+        )
+        .unwrap();
+
+        output
+    };
+
+    let decompressed = {
+        let mut output = vec![];
+        io::copy(
+            &mut read::Bz3Decoder::new(compressed.as_slice()).unwrap(),
```

Comment:
To make sure that this test is coherent, you can apply these diffs:
```diff
- &mut write::Bz3Encoder::new(&mut output, 100 * KB).unwrap(),
+ &mut xz2::write::XzEncoder::new(&mut output, 6),
```
and
```diff
- &mut read::Bz3Decoder::new(compressed.as_slice()).unwrap(),
+ &mut xz2::read::XzDecoder::new(compressed.as_slice()),
```

Then run this:
```sh
cargo add --dev xz2
cargo test -- test_compressing_and_decompressing_small_input
```

And the test will pass, I am assuming `xz2` is battle-tested and is behaving correctly right here.